### PR TITLE
add new provider Twsms (from platolin)

### PR DESCRIPTION
**Pull request by tubssp**

### CODE REVIEW

File: src/SmsSender/Provider/TwsmsProvider.php [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+use SmsSender\Exception as Exception;
+use SmsSender\HttpAdapter\HttpAdapterInterface;
+use SmsSender\Result\ResultInterface;
+/**
+ * Created by PhpStorm.
+ * User: plato
+ * Date: 2015/4/9
+ * Time: 下午 01:58
+ */
```
Comment (K-Phoen) [inline src/SmsSender/Provider/TwsmsProvider.php]: Could you remove this docblock?

**Comment by tubssp** [comment]
@K-Phoen block is removed

File: src/SmsSender/Provider/TwsmsProvider.php [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                'mobile' => $recipient,
+                'sendtime' => $sendtime,
+                'message' => $this->getMessage($body),
+            )
+        );
+
+        $extra_result_data = array(
+            'recipient' => $recipient,
+            'body' => $body,
+
```
Comment (K-Phoen) [inline src/SmsSender/Provider/TwsmsProvider.php]: Extra line here ;)

File: src/SmsSender/Provider/TwsmsProvider.php [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+                     * Account username (case sensitive)
+                     */
+                    'username' => $this->username,
+                    /*
+                     * Account password (case sensitive)
+                     */
+                    'password' => $this->password,
+                    /*
+                     * -- sendtime
+                     * 格式：YYYYMMDDHHII （請使用 24 小時制）
```
Comment (K-Phoen) [inline src/SmsSender/Provider/TwsmsProvider.php]: Could you translate these comments in english please?

File: src/SmsSender/Provider/TwsmsProvider.php [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+     * Parses the data returned by the API for a "send" request.
+     *
+     * @param string $result The raw result string.
+     * @param array $extra_result_data
+     *
+     * @return array
+     */
+    protected function parseSendResults($result, array $extra_result_data = array())
+    {
+
```
Comment (K-Phoen) [inline src/SmsSender/Provider/TwsmsProvider.php]: Extra line here.

File: src/SmsSender/Provider/TwsmsProvider.php [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+        $xml = simplexml_load_string($result);
+
+        if ($xml->code <> '00000') {
```
Comment (K-Phoen) [inline src/SmsSender/Provider/TwsmsProvider.php]: Could you use a *more standard* operator? Like `!=` or `!==` (if possible)

File: src/SmsSender/Provider/TwsmsProvider.php [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+     */
+    protected function checkForStatusResult($result)
+    {
+        if ('0#' === $result) {
+            return 'no message in queue';
+        }
+
+        $statuses = array(
+            1 => 'DELIVERED',
+            // Message delivered to handset.
```
Comment (K-Phoen) [inline src/SmsSender/Provider/TwsmsProvider.php]: The comment should be before the `1 => 'DELIVERED',`

File: src/SmsSender/Provider/TwsmsProvider.php [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+
+        return false;
+    }
+
+    /**
+     * @param  string $message
+     * @param  int $data_coding_scheme
+     * @return string
+     */
+    protected function getMessage($message, $data_coding_scheme = null)
```
Comment (K-Phoen) [inline src/SmsSender/Provider/TwsmsProvider.php]: You can remove this method and directly call `urlencode` instead.

**Comment by K-Phoen** [comment]
Thanks for this PR! :+1: 

Could you add tests? I'll merge it once it's done :)

**Comment by tubssp** [comment]
test come later.... please wait.

**Comment by K-Phoen** [comment]
ping @tubssp (or @platolin)

Could you provide tests? Without them, I won't merge this PR…
As I have no Twsms test account and no access to a english-written documentation, I can not write tests myself.

**Comment by tubssp** [comment]
I can not write test without @platolin

if @platolin not write tests then i close this PR.

**Pull request closed without merging**